### PR TITLE
Default message stringifier

### PR DESCRIPTION
**Pull request by IlliaShepilov**
Summary of changes:

* Created the default `Stringifier` for the `Message` classes.
* Moved the utility `builderFor` method to the `Messages` class.
* Created the utility method for the parsing `JSON` to the `Message` class.

The  usage of the`DefaultMessageStringifier` class:
```proto
// The `TaskId` Protobuf message definition.
message TaskId {
    string value = 1;
}
```
```proto
// The `TaskName` Protobuf message definition.
message TaskName {
    string value = 1;
}
```

```proto
// The `Task` Protobuf message definition.
message Task {
    TaskId id = 1;
    TaskName name = 2;
}
```
```java
// Construct the message.
final TaskId taskId = TaskId.newBuilder().setValue("task-id").build();
final TaskName taskName = TaskName.newBuilder().setValue("task-name").build();
final Task task = Task.newBuilder().setId(taskId).setTaskName(taskName).build();

// Obtain the default `Stringiifer`.
final Stringifier<Task> taskStringifier = StringifierRegistry.getStringifier(Task.class);

// The result is {"id":{"value":"task-id"},"name":{"value":"task-name"}}.
final String json = taskStringifier.reverse().convert(task);

// task.equals(taskFromJson) == true.
final Task taskFromJson = taskStringifier.convert(json);
```

### CODE REVIEW

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

**Comment by armiol** [comment]
@IlliaShepilov please add a code sample, illustrating the `Messsage`, its transformation to a `String` and the result. The same should go to the Javadoc section somewhere in the codebase.

File: client/src/main/java/org/spine3/base/Stringifiers.java [hunk 1]
```diff
@@ -266,4 +282,29 @@ protected Integer fromString(String s) {
                        .convert(s);
         }
     }
+
+    /**
+     * The default {@code Stringifier} for the {@code Message} classes.
```
Comment (armiol) [inline client/src/main/java/org/spine3/base/Stringifiers.java]: The sample should go here.

File: client/src/main/java/org/spine3/base/StringifierRegistry.java [hunk 2]
```diff
@@ -58,17 +59,27 @@ private StringifierRegistry() {
...
-            throw new MissingStringifierException(errMsg);
+        if (stringifierOptional.isPresent()) {
+            final Stringifier<T> stringifier = stringifierOptional.get();
+            return stringifier;
         }
-        final Stringifier<T> stringifier = stringifierOptional.get();
-        return stringifier;
+
+        if (typeOfT instanceof Class) {
+            final Class<?> aClass = (Class)typeOfT;
```
Comment (armiol) [inline client/src/main/java/org/spine3/base/StringifierRegistry.java]: The `isMessage` check should go to some utility class. Please check the existing utility classes in `spine-core`.

Comment (armiol) [inline client/src/main/java/org/spine3/base/StringifierRegistry.java]: I also think this line is not formatted.

File: client/src/main/java/org/spine3/base/StringifierRegistry.java [hunk 3]
```diff
@@ -58,17 +59,27 @@ private StringifierRegistry() {
         // Prevent external instantiation of this singleton class.
     }
 
+    @SuppressWarnings("unchecked") // It is OK because the class is checked before the cast.
```
Comment (armiol) [inline client/src/main/java/org/spine3/base/StringifierRegistry.java]: Can we extract the bit, where the unsafe cast is made, to a separate method and move the suppression to it?

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

**Comment by armiol** [comment]
@IlliaShepilov please include the original message declaration into both PR description and Javadoc.

**Comment by IlliaShepilov** [comment]
@armiol PTAL.

**Pull request merged**